components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -1131,11 +1148,37 @@ private class MonitoredItemWrapper {
                             EventTemplate.PendingAlarms.RetainIndex = eventFilter.SelectClauses.Count();
                             eventFilter.AddSelectClause(ObjectTypeIds.ConditionType, "Retain");
                         }
+
+                        // set up the timers
+                        if (EventTemplate.PendingAlarms.UpdateIntervalTimespan.HasValue) {
+                            _pendingAlarmsUpdateTimer.Interval = EventTemplate.PendingAlarms.UpdateIntervalTimespan.Value.TotalMilliseconds;
+                            _pendingAlarmsUpdateTimer.Elapsed += PendingAlarmsUpdateTimer_Elapsed;
+                            _pendingAlarmsUpdateTimer.Start();
+                        }
+                        if (EventTemplate.PendingAlarms.SnapshotIntervalTimespan.HasValue) {
+                            _pendingAlarmsSnapshotTimer.Interval = EventTemplate.PendingAlarms.SnapshotIntervalTimespan.Value.TotalMilliseconds;
+                            _pendingAlarmsSnapshotTimer.Elapsed += PendingAlarmsSnapshotTimer_Elapsed;
+                            _pendingAlarmsSnapshotTimer.Start();
+                        }
                     }
                     Item.Filter = eventFilter;
                 }
             }
 
+            private void PendingAlarmsUpdateTimer_Elapsed(object sender, System.Timers.ElapsedEventArgs e) {
```

Comment:
`OnPendingAlarmsUpdateTimerElapsed` would be a better name. To be consistent with the event handlers above.

Reply:
Agreed